shared/util/quit-helper.desktop.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+import electron from 'electron'
+import {ipcRenderer, ipcMain} from 'electron'
+import {quit} from '../../desktop/app/ctl'
+
+const app = electron.app || electron.remote.app
```

Comment:
We can get rid of `remote.app` usage here, but the we'll still need the BrowserWindow below.

We basically provide one function that will execute an array of actions, and that will send an ipc msg to run on the main thread or will run on the main thread itself (if called from the main thread).